### PR TITLE
Accept version changes in composed templates

### PR DESCRIPTION
**Pull request by turkenh**
Reflecting changes in https://github.com/crossplane/crossplane/pull/5369

### CODE REVIEW

File: render.go [hunk 1]
```diff
@@ -35,14 +34,16 @@ func RenderFromJSON(o resource.Object, data []byte) error {
...
+	// someone changed the kind or group in the template, or we're trying to use the
+	// wrong template (e.g. because the order of an array of anonymous templates
 	// changed).
-	empty := schema.GroupVersionKind{}
-	if gvk != empty && o.GetObjectKind().GroupVersionKind() != gvk {
-		return errors.Errorf(errFmtKindChanged, gvk, o.GetObjectKind().GroupVersionKind())
+	// Please note, we don't check for version changes, as versions can change. For example,
+	// if a composed resource was created with a template that has a version of "v1alpha1",
+	// and then the template is updated to "v1beta1", the composed resource will still be valid.
+	if !gvk.Empty() && o.GetObjectKind().GroupVersionKind().GroupKind() != gvk.GroupKind() {
```
Comment (phisco) [inline render.go:45]: afaict gvk here is always empty, should we take the chance to fix this too?

Comment (turkenh) [reply render.go:45]: Hmm, is `gvk` always emtpy? how about `o.GetObjectKind()` ? Wouldn't it always fail in the latter is not empty always?

Comment (phisco) [reply render.go:45]: If you look at [where](https://github.com/crossplane-contrib/function-patch-and-transform/blob/0d695806eb273a377abb9219abef2a768d906b99/fn.go#L140-L157) RenderFromJson is called, object is always an empty new Composed resource.

Comment (negz) [reply render.go:45]: Yeah, I think this whole function could just be `json.Unmarshal`.

**Pull request closed without merging**